thunderfish/fakefish.py

```diff
@@ -1,123 +1,233 @@
+"""
+Generate artificial waveforms of weakly electric fish.
+
+The two main functions are
+
+generate_wavefish()
+generate_pulsefish()
+
+for generating EODs of wave-type and pulse_type electric fish, respectively.
+
+The following functions use the two functions to generate waveforms of specific fishes:
+
+generate_alepto()
+
+generates a waveform mimicking the one of the wave-type fish Apteronotus leptorhynchus.
+
+generate_monophasic_pulses()
+generate_biphasic_pulses()
+generate_triphasic_pulses()
+
+generate waveforms of monphasic, biphasic and triphasic pulse-type fishes.
+"""
+
 import numpy as np
-from IPython import embed
 
-import matplotlib.pyplot as plt
 
+def generate_wavefish(frequency=100.0, samplerate=44100., duration=1., noise_std=0.01,
+                      amplitudes=1.0):
+    """Generate EOD of a wave-type fish.
 
-def generate_wavefish(freq, samplerate, time_len=20., harmonics=3):
-    """
+    The fundamental frequency of the EOD is given by frequency. The amplitude of the
+    fundamental is given by the first element in amplitudes. The amplitudes of harmonics
+    are give by optional further elements of the amplitudes list.
+
+    The generated waveform is duration seconds long and is sampled with samplerate Hertz.
+    Gaussian white noise with a standard deviation of noise_std is added to the generated
+    waveform.
 
-    :param freq: (float). Frequency of the fish in Hz.
+    :param frequency: (float). EOD frequency of the fish in Hz.
     :param samplerate: (float). Sampling rate in Hz.
-    :param time_len: (float). Length of the recording in sec.
-    :param harmonics: (int). Number of harmonics of freq to be included.
+    :param duration: (float). Duration of the generated data in seconds.
+    :param noise_std: (float). Standard deviation of additive Gaussian white noise.
+    :param amplitudes: (float or list of floats). Amplitudes of fundamental and optional harmonics.
 
-    :return: (array). Data of a wavefish with given frequency.
+    :return data: (array). Generated data of a wave-type fish.
     """
-    time = np.arange(0, time_len, 1./samplerate)
-
-    data = np.sin(2 * np.pi * time * freq)
-    for har in range(1, harmonics):
-        data += (np.sin(2*np.pi*time*(har*freq))) * 0.1*har
-
-    # Insert some noise
+    time = np.arange(0, duration, 1./samplerate)
+    data = np.zeros(len(time))
+    if np.isscalar(amplitudes):
+        amplitudes = [amplitudes]
+    for har, ampl in enumerate(amplitudes):
```

Comment:
I don't understand what you mean with amplitudes... but perhaps a live explanation is more successful.